File: frontend/src/app/models/book.ts
import { Loan } from './loan';

export interface Book {
  _id?: string;
  title: string;
  author: string;
  year: number;
  genre: string;
  available: boolean;
  hasActiveLoan?: boolean;
  activeLoan?: Loan | null;
}

export interface BookFilters {
  title?: string;
  genre?: string;
  author?: string;
  available?: boolean | '' | null;
}

File: borrow-return_backend/source/Domain/Book.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace Domain;

public record Book
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string? Id { get; init; }

    public required string Title { get; init; }

    public required string Author { get; init; }

    public required int Year { get; init; }

    public required string Genre { get; init; }

    public bool Available { get; init; } = true;
}
